qcodes/dataset/data_export.py

```diff
@@ -69,11 +69,16 @@ def get_data_by_id(run_id: int) -> \
 
     """
     ds = load_by_id(run_id)
+    output = _get_data_from_ds(ds)
+    return output
+
 
+def _get_data_from_ds(ds: DataSet) -> List[List[Dict[str, Union[str, np.ndarray]]]]:
     dependent_parameters: Tuple[ParamSpecBase, ...] = ds.dependent_parameters
 
-    parameter_data = ds.get_parameter_data(
-        *[ps.name for ps in dependent_parameters])
+    all_data = ds.cache.data()
+
+    parameter_data = {ps.name: all_data[ps.name] for ps in dependent_parameters}
```

Comment:
This is not the same behavior when `dependent_parameters == ()` as get_parameter_data treats `()` as `None` which means all parameters